server/src/server.rs

```diff
@@ -167,7 +168,7 @@ where
 			}
 		}
 
-		connections.await;
+		while connections.next().await.is_some() {}
```

Comment:
Ah ok, so if I understand this right, `FuturesUnordered` won't poll anything until you tell it to, so you are spawning tasks into it and then this line just makes sure that all tasks are completed before the function ends?

Reply:
yeah, exactly